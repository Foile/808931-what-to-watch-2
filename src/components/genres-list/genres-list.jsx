import React from "react";
import PropTypes from 'prop-types';

const GenresList = ({genres, onGenreClick}) =>
  (<ul className="catalog__genres-list">
    { genres.map((genre)=>
      (<li className="catalog__genres-item" key={`catalog__genres-item--${genre}`} onClick={(evt)=>{
        evt.preventDefault();
        onGenreClick(genre);
      }}>
        <a href="#" className="catalog__genres-link" id={genre}>
          {genre}
        </a>
      </li>))}</ul>);

GenresList.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  onGenreClick: PropTypes.func.isRequired
};

export default GenresList;
