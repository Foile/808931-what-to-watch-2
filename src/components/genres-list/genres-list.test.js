import React from 'react';
import renderer from "react-test-renderer";
import {GenresList} from './genres-list';

it(`App correctly renders after relaunch`, () => {
  const genres = [`Comedy`, `Drama`];
  const tree = renderer
  .create(<GenresList genres = {genres} onGenreClick={jest.fn()}/>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
});
