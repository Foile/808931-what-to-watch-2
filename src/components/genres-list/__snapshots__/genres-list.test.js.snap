// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders after relaunch 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
    onActiveChange={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
      id="Comedy"
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item"
    onActiveChange={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
      id="Drama"
    >
      Drama
    </a>
  </li>
</ul>
`;
